src/DatasourceWithAsyncBackend.ts

```diff
@@ -136,7 +136,8 @@ export class DatasourceWithAsyncBackend<
           };
 
           let headers = {};
-          if (isRunning(status)) {
+          const cachingDisabled = !config.featureToggles.useCachingService || !config.featureToggles.awsAsyncQueryCaching
```

Comment:
Two questions:
* How come we need to involve config.featureToggles.awsAsyncQueryCaching here? What is it about the original code that had anything to do with config.featureToggles.awsAsyncQueryCaching?
* What do you think about including `isRunning` in the determination of `cachingDisabled` in order to simplify `if (cachingDisabled && isRunning(status))`?

Reply:
There might be a better way to do this, but basically I needed to add an additional check to the query service to get it to work with async aws queries (see the linked pr in the description), and without that additional check we'll end up caching "running" states as well. Technically a user could enable this flag when the fix isn't present and break their version of the plugin, but this is a bit of a guard for if they have the caching service enabled and a version of the plugin that has this check, but aren't on a version of main grafana with that check added.